processor/src/trace/utils.rs

```diff
@@ -190,10 +191,63 @@ impl ChipletsLengths {
     }
 }
 
+// AUXILIARY COLUMN BUILDER
+// ================================================================================================
+
+/// Defines a builder responsible for building a single column in an auxiliary segment of the
+/// execution trace.
+pub trait AuxColumnBuilder<E: FieldElement<BaseField = Felt>> {
+    // REQUIRED METHODS
+    // --------------------------------------------------------------------------------------------
+
+    fn get_requests_at(&self, main_trace: &MainTrace, alphas: &[E], row_idx: usize) -> E;
+
+    fn get_responses_at(&self, main_trace: &MainTrace, alphas: &[E], row_idx: usize) -> E;
+
+    // PROVIDED METHODS
+    // --------------------------------------------------------------------------------------------
+
+    fn init_requests(&self, _main_trace: &MainTrace, _alphas: &[E]) -> E {
+        E::ONE
+    }
+
+    fn init_responses(&self, _main_trace: &MainTrace, _alphas: &[E]) -> E {
+        E::ONE
+    }
+
+    /// Builds the chiplets bus auxiliary trace column.
+    fn build_aux_column(&self, main_trace: &ColMatrix<Felt>, alphas: &[E]) -> Vec<E> {
+        let main_trace = MainTrace::new(main_trace);
+        let mut responses_prod: Vec<E> = unsafe { uninit_vector(main_trace.num_rows()) };
+        let mut requests_prod: Vec<E> = unsafe { uninit_vector(main_trace.num_rows()) };
+        responses_prod[0] = self.init_responses(&main_trace, alphas);
+        requests_prod[0] = self.init_requests(&main_trace, alphas);
+
+        let mut result_2_acc = E::ONE;
```

Comment:
I would rename to `requests_running_prod` (to differentiate from `responses_prod`, which is a vector of running products)

Reply:
Updated